trl/trainer/sft_trainer.py

```diff
@@ -401,7 +401,7 @@ def __init__(
                 model, processing_class = clone_chat_template(model, processing_class, args.chat_template_path)
 
         # PEFT configuration and model wrapping
-        if peft_config is not None:
+        if peft_config is not None or (is_peft_available() and isinstance(model, PeftModel)):
```

Comment:
is this needed? as `_prepare_peft_model` checks for these two conditions and either returns an error or returns back the model?